packages/ui/Transition/Transition.js

```diff
@@ -527,6 +526,19 @@ class Transition extends React.Component {
 		}
 	}
 
+	setRegistry = registry => {
```

Comment:
This might not be necessary here. When using `contextType` and the lifecycle methods, we know that `register`/`unregister` are only invoked a mount/unmount. With render props, you need a way to only register the first time which is where this helps a bit.

After typing this comment, it occurs to me that when using render props we might be able to register with: ```
<ResizeContext.Subscriber>
  {value => {
    if (!this.registry) {
      this.registry = value;
      this.registry.register(this.handleResize);
    }
    ...
  }
</ResizeContext.Consumer>
```
and unregister in `componentWillUnmount` with `if (this.registry) this.registry.unregister(this.handleResize)`

Reply:
Yeah that makes sense. I've simplified them now.